setup.py

```diff
@@ -79,17 +57,82 @@ def install_triton_client():
             f.write(response.content)
         response.close()
         tar_file = tarfile.open(save_name, mode='r')
-        tar_file.extractall()
+        tar_file.extract(f'python/{wheel_file}', path=save_name.parent)
         tar_file.close()
 
-    package_path = save_name.parent / f'python/tensorrtserver-{TRITON_CLIENT_VERSION}-py2.py3-none-linux_x86_64.whl'
+    package_path = save_name.parent / f'python/{wheel_file}'
 
     return package_path
 
 
+# get torch ,torchvision and tensorflow version
+# reference: https://github.com/openvinotoolkit/nncf/blob/develop/setup.py
+# reference: https://www.tensorflow.org/install/source#gpu
+PRECOMPILED_TENSORFLOW_PAIRS = {
+    "cu110": {
+        "tensorflow": "tensorflow-gpu==2.4.0",
+        "tensorflow-serving-api": "tensorflow-serving-api-gpu==2.4.0"
+    },
+    "cu102": {
+        "tensorflow": "tensorflow-gpu==2.3.0",
+        "tensorflow-serving-api": "tensorflow-serving-api-gpu==2.3.0"
+    },
+    "cu101": {
+        "tensorflow": "tensorflow-gpu==2.3.0",
+        "tensorflow-serving-api": "tensorflow-serving-api-gpu==2.3.0"
+    },
+    "cpu": {
+        "tensorflow": "tensorflow==2.3.0",
+        "tensorflow-serving-api": "tensorflow-serving-api==2.3.0"
+    }
+}
+
+CUDA_VERSION = "cpu"
+PYTHON_VER = f'{sys.version_info[0]}{sys.version_info[1]}'
+torch_cuda_version = "cpu"
+tensorflow_cuda_version = "cpu"
+
+
+if "CUDA_HOME" in os.environ:
+    cuda_version_file = os.path.join(os.environ["CUDA_HOME"], "version.txt")
+    if os.path.exists(cuda_version_file):
+        with open(cuda_version_file) as f:
+            CUDA_VERSION = "cu".join(f.readline().strip().split(" ")[-1].split(".")[:2])
+            if CUDA_VERSION not in ["cpu", "cu92", "cu101", "cu102"]:
```

Comment:
So how many versions of CUDA do we support now, I saw 92, 101 and 102 here but for tensorflow installation I saw 101, 102 and 110.

Reply:
For torch==1.5.0, only CUDA version 9.2,10.1 and 11.0 can install precompiled GPU packages, for tensorflow==2.3.0, only 10.1, 10.2  and 11.0 are supported(BTW, TensorFlow 2.4.0 seems not working in modelci).

Reply:
Can you figure out why TF 2.4.0 not working?

Reply:
I came across this issue several times: https://github.com/tensorflow/tensorflow/issues/38012, but 2.4.0 did work this time